#!/usr/bin/env python
PACKAGE = "controller_tutorial"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("gain_p",                   double_t, 0, "A double parameter", 1.0, 0, 5)
gen.add("gain_i",                   double_t, 0, "A double parameter", 0.1, 0, 2)
gen.add("gain_d",                   double_t, 0, "A double parameter", 0.001, 0, 0.1)
gen.add("stop_margin_angle",        double_t, 0, "A double parameter", 0.1, 0, 0.2)
gen.add("stop_margin_distance",     double_t, 0, "A double parameter", 0.03, 0, 0.3)
gen.add("max_linear_velocity",      double_t, 0, "A double parameter", 0.5, 0, 2)
gen.add("max_angular_velocity",     double_t, 0, "A double parameter", 0.6, 0, 5)

exit(gen.generate(PACKAGE, "controller_test", "controller_test"))